apps/files_sharing/lib/Controller/ShareesAPIController.php

```diff
@@ -257,7 +257,7 @@ private function getAllSharees(string $user, array $shareTypes): ISearchResult {
 			$sharees = $this->getAllShareesByType($user, $shareType);
 			$shareTypeResults = [];
 			foreach ($sharees as [$sharee, $displayname]) {
-				if (!isset($this->searchResultTypeMap[$shareType])) {
+				if (!isset($this->searchResultTypeMap[$shareType]) || empty($sharee)) {
```

Comment:
Will break on valid `'0'` value I think, as it’s a valid user id but return true on empty check.
```suggestion
				if (!isset($this->searchResultTypeMap[$shareType]) || ($sharee === '')) {
```

Reply:
(do we know why we have shares with empty `getSharedWith` return? Is that normal behavior?)

Reply:
yes, it can happens for link shares for example :)
Or mail shares without  recipient yet